modules/gdscript/gdscript_editor.cpp

```diff
@@ -936,6 +936,11 @@ static void _find_annotation_arguments(const GDScriptParser::AnnotationNode *p_a
 		}
 	} else if (p_annotation->name == SNAME("@warning_ignore")) {
 		for (int warning_code = 0; warning_code < GDScriptWarning::WARNING_MAX; warning_code++) {
+#ifndef DISABLE_DEPRECATED
+			if (warning_code >= GDScriptWarning::PROPERTY_USED_AS_FUNCTION) {
```

Comment:
Probably warrants a comment in the enum definition so it's clear to future contributors that if they add a deprecated warning code before `PROPERTY_USED_AS_FUNCTION`, this code should be updated here.

Or alternatively, add an alias in the enum (after `WARNING_MAX`?) that would be
```cpp
DEPRECATED_WARNINGS = PROPERTY_USED_AS_FUNCTION, // Should point to the first deprecated enum value.
```